WebHook/WebHookFunction.cs

```diff
@@ -0,0 +1,162 @@
+            });
+
+            return "truth";
+        }
+
+        private static async Task<string> ProcessInstallationAsync(Hook hook, ICollector<RouterMessage> routerMessages, CloudTable installationTable)
+        {
+            switch (hook.action)
+            {
+                case "created":
+                    foreach (var repo in hook.repositories)
+                    {
+                        routerMessages.Add(new RouterMessage
+                        {
+                            InstallationId = hook.installation.id,
+                            Owner = hook.installation.account.login,
+                            AccessTokensUrl = hook.installation.access_tokens_url,
+                            RepoName = repo.name,
+                            CloneUrl = $"https://github.com/{repo.full_name}",
+                        });
+                    }
+
+                    break;
+                case "added":
+                    foreach (var repo in hook.repositories_added)
+                    {
+                        routerMessages.Add(new RouterMessage
+                        {
+                            InstallationId = hook.installation.id,
+                            Owner = hook.installation.account.login,
+                            AccessTokensUrl = hook.installation.access_tokens_url,
+                            RepoName = repo.name,
+                            CloneUrl = $"https://github.com/{repo.full_name}",
+                        });
+                    }
+
+                    break;
+                case "removed":
+                    foreach (var repo in hook.repositories_removed)
+                    {
+                        await installationTable.DropRow(hook.installation.id.ToString(), repo.name);
+                    }
+
+                    break;
+                case "deleted":
+                    await installationTable.DropPartitionAsync(hook.installation.id.ToString());
+                    break;
+            }
+
+            return "truth";
+        }
+
+        private static async Task<string> ProcessMarketplacePurchaseAsync(Hook hook, CloudTable marketplaceTable)
+        {
+            switch (hook.action)
+            {
+                case "purchased":
+                    await marketplaceTable.ExecuteAsync(TableOperation.InsertOrMerge(new Marketplace(hook.marketplace_purchase.account.id, hook.marketplace_purchase.account.login)
+                    {
+                        AccountType = hook.marketplace_purchase.account.type,
+                        PlanId = hook.marketplace_purchase.plan.id,
+                        SenderId = hook.sender.id,
+                        SenderLogin = hook.sender.login,
+                    }));
+
+                    return "purchased";
+                case "cancelled":
+                    await marketplaceTable.DropRow(hook.marketplace_purchase.account.id.ToString(), hook.marketplace_purchase.account.login);
```

Comment:
Looks a bit odd that you need the `ToString` in the delete but not in the insert? Maybe you could do that inside the `DropRow` function?